### PR TITLE
Paramters object allowing updates after first run

### PR DESCRIPTION
**Pull request by mjkoo**
Breaking change

### CODE REVIEW

File: src/polyswarmclient/bountiesclient.py [hunk 1]
```diff
@@ -107,7 +110,7 @@ def __init__(self, client):
             'uri': artifact_uri,
             'duration': duration,
         }
-        bounty_fee = self.parameters[chain]['bounty_fee']
+        bounty_fee = self.parameters[chain].get('bounty_fee')
```
Comment (Rizato) [inline src/polyswarmclient/bountiesclient.py]: Should this be awaited?

File: src/polyswarmclient/bountiesclient.py [hunk 2]
```diff
@@ -160,7 +163,7 @@ def __init__(self, client):
             'mask': mask,
             'commitment': commitment,
         }
-        fee = self.parameters[chain]['assertion_fee']
+        fee = self.parameters[chain].get('assertion_fee')
```
Comment (Rizato) [inline src/polyswarmclient/bountiesclient.py]: Same with this one

File: src/polyswarmclient/bountiesclient.py (hunk 1 above)
Comment (mjkoo) [reply src/polyswarmclient/bountiesclient.py]: ya ty

File: src/polyswarmclient/bountiesclient.py [hunk 3]
```diff
@@ -67,7 +68,7 @@ def has_required_event(self, transaction_events):
         for assertion in assertions:
             if (assertion.get('bid', 0) == self.bid and
                     assertion.get('mask', []) == self.mask and
-                    assertion.get('nonce', '') == self.nonce and
+                    assertion.get('commitment', '') == self.commitment and
```
Comment (mjkoo) [inline src/polyswarmclient/bountiesclient.py]: @Rizato rebased off current master, in resolving changed this to commitment, pretty sure thats what we want here. Thoughts?

Comment (Rizato) [reply src/polyswarmclient/bountiesclient.py]: Yea, that does make sense.

**Pull request merged**